armi/reactor/composites.py

```diff
@@ -3032,15 +3031,25 @@ def _getReactionRates(self, nucName, nDensity=None):
         -----
         If you set nDensity to 1/CM2_PER_BARN this makes 1 group cross section generation easier
         """
+        from armi.reactor.assemblies import Assembly
         from armi.reactor.blocks import Block
+        from armi.reactor.reactors import Core
 
         if nDensity is None:
             nDensity = self.getNumberDensity(nucName)
+
         try:
+            if isinstance(self, (Assembly, Core)):
+                block = self.getChildren(
+                    deep=True, predicate=lambda o: isinstance(o, Block)
+                )[0]
+            else:
+                block = self.getAncestor(lambda x: isinstance(x, Block))
+
             return getReactionRateDict(
                 nucName,
-                self.getAncestorWithFlags(Flags.CORE).lib,
-                self.getAncestor(lambda x: isinstance(x, Block)).getMicroSuffix(),
+                self.getAncestor(lambda c: isinstance(c, Core)).lib,
+                block.getMicroSuffix(),
```

Comment:
I think that there is a problem here. If the object passed into `_getReactionRates()` is an `Assembly` or `Core`, then the `block` used here will just be some random block, and the micro suffix will be blanket applied to all of the reaction rate calculations. This is not correct, because in reality each block in a core can have a different suffix (and thus different XS set).

Let me know if you disagree or don't follow.

Reply:
Ah, maybe you're right.  Okay, I got a little fast-and-loose.

But we don't define micro-suffixes for the Core or Reactor level.

Reply:
> But we don't define micro-suffixes for the Core or Reactor level

Agreed, but when looping over the blocks in the core, the reaction rate calculation for each block should use the suffix for that specific block.

I was trying to think of how to implement it correctly based on your changes, and I anticipate you'll have to do a little refactoring of the method calls.

Reply:
Okay, okay, my solution is bonk.

Should the reaction rates in the Core be the sum of the Reaction rates of all the Blocks in the Core?  If that's the case, I guess I can figure something out.

I suppose I could also just have `Core.getReactionRates()` throw an error, but I'd like to think we're not there yet.

Reply:
Yes, the reaction rates returned by `getReactionRates()` are said to be volume integrated, so just summing them all up is what you're looking to accomplish. 

I don't think its necessary to throw an error on a Core object. This seems to be doable, just with a little reformulation. 

In particular, I think that if the call to `getChildren()` [here](https://github.com/terrapower/armi/blob/36a5bdfde7c4ef32d079225b1938e5db86618276/armi/reactor/composites.py#L3086) were replaced with `getChildren(deep=True)`, then we could guarantee that the objects passed into `_getReactionRates()` are always `Component`s, which always are associated with a given micro suffix of their parent blocks.

Do you think that would work?

Reply:
Okay, take a look now.  Tell me what you think.

Sorry for the futz, perhaps I have too many balls in the air today.

Reply:
Yep, I think this works. I made another suggestion to simplify, but in general I think this is looking correct now.

https://github.com/terrapower/armi/pull/1771#discussion_r1674190476